File: RelaxPluginDemo/app/src/main/java/com/ustory/relax_business_component/plugin/inter/IPlugin.java
package com.ustory.relax_business_component.plugin.inter;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.KeyEvent;
import android.view.Menu;
import android.view.MenuItem;
import android.view.MotionEvent;
import android.view.ViewGroup;

/**
 * Created by qiyue on 2018/1/31.
 */

public interface IPlugin {

    public void onCreate(Bundle savedInstanceState);
    public void onStart();
    public void onRestart();
    public void onActivityResult(int requestCode, int resultCode, Intent data);
    public void onResume();
    public void onPause();
    public void onStop();
    public void onDestroy();
    public void attach(Activity proxyActivity);
    public void onSaveInstanceState(Bundle outState);
    public void onNewIntent(Intent intent);
    public void onRestoreInstanceState(Bundle savedInstanceState);
    public boolean onTouchEvent(MotionEvent event);
    public boolean onKeyUp(int keyCode, KeyEvent event);
    public void onWindowAttributesChanged(ViewGroup.LayoutParams params);
    public void onWindowFocusChanged(boolean hasFocus);
    public void onBackPressed();
    public boolean onCreateOptionsMenu(Menu menu);
    public boolean onOptionsItemSelected(MenuItem item);

    public void startPluginActivity(String ClassName);


}

File: relax-business-component/src/main/java/com/ustory/relax_business_component/plugin/inter/IPlugin.kt
package com.ustory.relax_business_component.plugin.inter

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.*

/**
 * Created by qiyue on 2018/1/31.
 */

interface IPlugin {

    fun onCreate(savedInstanceState: Bundle)
    fun onStart()
    fun onRestart()
    fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent)
    fun onResume()
    fun onPause()
    fun onStop()
    fun onDestroy()
    fun attach(proxyActivity: Activity)
    fun onSaveInstanceState(outState: Bundle)
    fun onNewIntent(intent: Intent)
    fun onRestoreInstanceState(savedInstanceState: Bundle)
    fun onTouchEvent(event: MotionEvent): Boolean
    fun onKeyUp(keyCode: Int, event: KeyEvent): Boolean
    fun onWindowAttributesChanged(params: ViewGroup.LayoutParams)
    fun onWindowFocusChanged(hasFocus: Boolean)
    fun onBackPressed()
    fun onCreateOptionsMenu(menu: Menu): Boolean
    fun onOptionsItemSelected(item: MenuItem): Boolean

    fun startPluginActivity(ClassName: String)


}
